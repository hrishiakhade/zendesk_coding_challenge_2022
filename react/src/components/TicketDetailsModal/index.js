import React from "react";
import Modal from "react-modal";
import "../../App.css";
import { formatDate } from "../../utils/common-helpers";
import strings from "../../utils/constants/strings";
// Modal.setAppElement("#root");

const customStyles = {
  content: {
    top: "50%",
    left: "50%",
    right: "auto",
    bottom: "auto",
    marginRight: "-50%",
    transform: "translate(-50%, -50%)",
    border: ".5px dotted lightgray",
    borderRadius: "10px",
    // minWidth: "80%",
    // minHeight: "60%",
    maxHeight: "600px",
    maxWidth: "800px",
    zindex: 99,
    boxShadow: '0px 8px 15px rgba(207, 207, 207, 0.7)',
  },
};

const TicketDetailsModal = (props) => (
  <div>
    <Modal
      isOpen={props?.selectedTicket ? true : false}
      contentLabel={`${strings.ticket_hash}${props?.selectedTicket?.id} Details`}
      onRequestClose={props.resetSelectedTicket}
      closeTimeoutMS={200}
      style={customStyles}
    >
      <div className="m-3">
        <span onClick={() => props.resetSelectedTicket()} className="cross">
          x
        </span>
        <h4>
          {strings.ticket_hash}
          {props?.selectedTicket?.id + " " + props?.selectedTicket?.subject}
        </h4>
        {props?.selectedTicket && (
          <div>
            <div className="d-flex flex-row">
              <strong className="text-left">
                {strings.req_id}
                {props.selectedTicket.requester_id}
              </strong>
              <strong className="text-left ml-auto">
                {strings.assign_id}
                {props.selectedTicket.assignee_id}
              </strong>
            </div>
            <div className="d-flex flex-row">
              <span className="date text-left" style={{paddingTop:'3px'}}>
                {formatDate(props.selectedTicket.created_at)}
              </span>
              <span className={`status-pill text-left ml-auto ${props.selectedTicket.status=='close' ? 'close' : props.selectedTicket.status=='pending' ? 'pending' : 'open'}`}>
                {" "}
                {
                  props.selectedTicket.status.charAt(0).toUpperCase() +
                  props.selectedTicket.status.slice(1)}
              </span>
            </div>
            <br />
            <p>{props.selectedTicket.description}</p>
          </div>
        )}
      </div>
    </Modal>
  </div>
);

export default TicketDetailsModal;
