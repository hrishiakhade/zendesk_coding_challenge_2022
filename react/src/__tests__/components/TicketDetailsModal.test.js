import React from 'react';
import Enzyme, { shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJSON from 'enzyme-to-json'
import TicketDetailsModal from '../../components/TicketDetailsModal';
import ticketData from '../../utils/fixtures/ticketData'

Enzyme.configure({
  adapter: new Adapter()
})

test('Should render TicketDetailsModal correctly with props', () => {
  const wrapper = shallow(<TicketDetailsModal selectedTicket={ticketData} />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
