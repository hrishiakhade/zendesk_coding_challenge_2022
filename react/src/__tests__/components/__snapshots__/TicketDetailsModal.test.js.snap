// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render TicketDetailsModal correctly with props 1`] = `
<div>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={200}
    contentElement={[Function]}
    contentLabel="Ticket#95 Details"
    isOpen={true}
    overlayElement={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    preventScroll={false}
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "border": ".5px dotted lightgray",
          "borderRadius": "10px",
          "bottom": "auto",
          "boxShadow": "0px 8px 15px rgba(207, 207, 207, 0.7)",
          "left": "50%",
          "marginRight": "-50%",
          "maxHeight": "600px",
          "maxWidth": "800px",
          "right": "auto",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "zindex": 99,
        },
      }
    }
  >
    <div
      className="m-3"
    >
      <span
        className="cross"
        onClick={[Function]}
      >
        x
      </span>
      <h4>
        Ticket#
        95 ex cillum cillum proident commodo
      </h4>
      <div>
        <div
          className="d-flex flex-row"
        >
          <strong
            className="text-left"
          >
            Requester ID: 
            1267068077489
          </strong>
          <strong
            className="text-left ml-auto"
          >
            Assignee ID: 
            1267068077489
          </strong>
        </div>
        <div
          className="d-flex flex-row"
        >
          <span
            className="date text-left"
            style={
              Object {
                "paddingTop": "3px",
              }
            }
          >
             Nov 21 2021
          </span>
          <span
            className="status-pill text-left ml-auto open"
          >
             
            Open
          </span>
        </div>
        <br />
        <p>
          Culpa amet mollit esse tempor sit ullamco sit ex dolore nostrud consectetur. Amet ad aliquip aliqua aute quis sunt laborum excepteur occaecat. Enim cillum do reprehenderit cupidatat ex eu ea ipsum ullamco consequat dolore duis consectetur. Magna sit voluptate laboris nisi do.

Dolor voluptate anim consequat ad cillum excepteur nisi elit. Ea tempor ipsum sint fugiat dolor anim amet pariatur aliqua labore ipsum. Sunt cillum eu fugiat est cillum mollit ullamco dolor anim in cupidatat cupidatat. Commodo eiusmod est et dolore enim do Lorem sunt quis magna sunt excepteur. Aliqua esse velit nulla labore culpa reprehenderit magna cupidatat aliquip. Do aliquip elit adipisicing aute excepteur culpa. Ullamco elit enim duis ut Lorem proident qui cupidatat magna.
        </p>
      </div>
    </div>
  </Modal>
</div>
`;
